parso/tree.py

```diff
@@ -20,12 +18,12 @@ def search_ancestor(node, *node_types):
             return node
 
 
-class NodeOrLeaf(object):
+class NodeOrLeaf:
     """
     The base class for nodes and leaves.
     """
     __slots__ = ()
-    type = None
+    type: str
```

Comment:
Is there a code path where `type` is None? (I guess not if you said solely `str` for the annotation?)

Reply:
No it shouldn't be.